test/e2e/run.sh

```diff
@@ -23,27 +23,47 @@ export cni_plugin=${cni_plugin:-cilium}
 export cni_release=${cni_release:-latest}
 TOPOLOGY_DIR=${TOPOLOGY_DIR:=e2e}
 
-K8S_REPO="https://github.com/kubernetes/kubernetes"
+GH_K8S_REPO="kubernetes/kubernetes"
 export k8s_release=${k8s_release:-"latest"}
 export k8s_version=""
 
 source "$LIB_DIR"/vm.bash
 
+list-github-releases () {
+    local repo="$1" max="${max_releases:-20}" page=1 tags="" more="" sep=""
+    local total=0 cnt=-1
+
+    while [ "$cnt" != 0 -a "$total" -lt "$max" ]; do
+        echo -n "fetching releases for $repo (page $page)..." 1>&2
+        if more=$(wget -q "https://github.com/$repo/releases?page=$page" -O- | \
+                      grep -E '.*/releases/tag/v[0-9]+.[0-9]+.[0-9]+"' | \
+                      sed -E 's|.*/releases/tag/(v[0-9]+.[0-9]+.[0-9]+).*|\1|g')
+        then
+            tags="$tags$sep$more"
+            sep=" "
+            cnt="$(echo $more | wc -w)"
+            total=$((total+cnt))
+            echo " got $cnt more, $total in total..." 1>&2
+            let page=$page+1
+        else
+            echo "failed to determine latest github release for $repo"
+            return 1
+        fi
+    done
+
+    echo "$tags" | tr -s ' ' '\n' | sort -Vr
+}
+
 latest-github-release () {
-    local repo="$1"
-    local latest=$(wget -q -O- $repo/releases/latest | grep '/tree/v[0-9.]*' |
-                       sed -E 's:.*/tree/v([0-9]*\.[0-9]*)(\.[0-9]*).*:\1\2:g' | head -1)
-    if [ -z "$latest" ]; then
-        echo "failed to determine latest release of $repo"
-        exit 1
-    fi
-    echo "$latest"
+    list-github-releases "$1" | head -1
```

Comment:
Given that `latest-github-release` is digging out only the latest release, would it suffice if it calls `list-github-releases` with max_releases=1 or max_releases=2?

Reply:
Unfortunately no. Generally speaking this whole thing is probabilistic. We don't know how many releases are listed per page (depends on how much collective metadata is associated with each release), and there is no guarantee that the latest release would be among the ones listed on the first page. So we're now a bit conservative and rather list too many than too few.